pkg/api/networkservice/connection_helpers.go

```diff
@@ -157,6 +157,48 @@ func (m *Connection) MatchesMonitorScopeSelector(selector *MonitorScopeSelector)
 	return false
 }
 
+// GetCurrentPathSegment - Get the current path segment of the connection
+func (m *Connection) GetCurrentPathSegment() *PathSegment {
+	if m == nil {
+		return nil
+	}
+	if m.GetPath().GetPathSegments() == nil {
```

Comment:
is it possible that GetPath could return nil?

Reply:
GRPC protects against nil receivers :)  if the receiver (m.GetPath()) is nil, GetPathSegments() will properly handle that and also return nil :)